extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxCurrentContextFactory.java

```diff
@@ -39,7 +39,13 @@ public void set(T state) {
             Context context = Vertx.currentContext();
             if (context != null && VertxContext.isDuplicatedContext(context)) {
                 VertxContextSafetyToggle.setContextSafe(context, true);
-                context.putLocal(LOCAL_KEY, state);
+                // this is racy but should be fine, because DC should not be shared
+                // and never remove the existing mapping
```

Comment:
I'm thinking about this and I was able to convince myself that it's true :-)

The only case when `oldState != state` could be when `oldState == null`. A single duplicated context can be easily shared between _threads_, but cannot be shared between _requests_, which is what matters here.

Reply:
I got some prior statements on the code (doesn't mean that is true eh): https://github.com/quarkusio/quarkus/blob/2e5c007ea2441ec9c4aaedc2e094f6a0dd4b966c/extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxCurrentContextFactory.java#L52-L53

We should remove that one as well if we know that's not a valid statement.
In relation of the changes here (comment apart, that could be removed), if we have racy writes we're still at the mercy of casual ordering of acquisition of the the CHM segment "lock", which means...random.
In short, this PR shouldn't mess up with that, in theory.

Reply:
No, that's true. Basically what I had to figure out for myself is what the comments mean when they say "should not be shared". It is entirely possible for the DCs to be shared among _threads_, but they should not be shared among _requests_. It probably took me longer than it should have, but it finally clicked :-)

Reply:
> but they should not be shared among requests

Exactly true

Reply:
> It is entirely possible for the DCs to be shared among threads, but they should not be shared among requests.

But it still raise the question: 2 threads sharing such would end up competing to write different `state` instances for the same `LOCAL_KEY`? 
My answer is, probably not, if the first time is set/activated isn't racy, instead, while subsequent ones, yep.
The request ctx state than, should be the same on both threads who share it and it just means that my "racy check" would just save replacing the same value (which imply avoiding costy CHM::put over and over).

Reply:
Right, that's also my conclusion. If 2 threads compete for writing to the same duplicated context, they should be both trying to write the same request context state.

Reply:
If that's true I would expect that is never racy at this point, because after the initial setting up, value will be stable (is not removed).
Now another question...

```
VertxContextSafetyToggle.setContextSafe(context, true);
```
This is still performing few checks if the context is duplicated bla bla...at this point I could shield calling it based on LOCAL_KEY's value being the same or not and save doing [these checks](https://github.com/quarkusio/quarkus/blob/53a64eaa2034375fcee588329b3450cbc41df5af/extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/context/VertxContextSafetyToggle.java#L119) over and over (although it is internally shielded, see https://github.com/quarkusio/quarkus/pull/30024)?